### PR TITLE
feat(commands): implement RFC 7151 (File Transfer Protocol HOST Command for Virtual Hosts)

### LINKED ISSUE #114: Any way to detect domain ?
If someone logs in using the domain name, like `mysite.com:21`, is there a way to find that out within the server ?

I noticed there's a `domain` property in the data object passed in the login event, but it's always null.

### PR DESCRIPTION
**Pull request by lezsakdomi**
Fix #114 

Todo:
  - [x] Add handler for command
  - [x] Add listener
  - [x] Provide option for checking hostname just at login time
  - [ ] Checking for hostname (domain name / ip literal) syntax according to [section 3.1](https://tools.ietf.org/html/rfc7151#section-3.1), including port number persistence
  - [ ] Checking if IP address matches server IP (in case supplied) ([section 3.1](https://tools.ietf.org/html/rfc7151#section-3.1))
  - [ ] When TLS used, checking if supplied hostname matches with one from TLS ([section 3.2.2](https://tools.ietf.org/html/rfc7151#section-3.2.2))
  - [ ] CLI options (specifying a map between virtualhosts and user/pass/root combos)
  - [ ] Tests

### CODE REVIEW

**Comment by lezsakdomi** [comment]
**[BUG]** It looks like passing `motd` and/or `anonymous` has no effect :(

**Comment by trs** [comment]
Thanks for the start of this PR! I'll take a look into giving you a hand with in when I get a chance. I'll read up on the `HOST` command first.

**Comment by lezsakdomi** [comment]
TL;DR:

The parameter of a host command is the hostname, obviously.
The hostname SHOULD NOT contain a port.
If it's an IPv6, the hostname SHOULD be encapsulated in `[]`s. If an IPv4, it MAY be enclosed between brackets or provided in the IPv6 form.

According to the specs, the server should:
- Validate the syntax (I was afraid to implement this, because validating IPv6 addresses is hard)
- Validate hostname validity if possible
  - If it's a bare IP (the IP should be our public IP)
  - If TLS used and hostname was in certificate

**Comment by lezsakdomi** [comment]
I think this project shouldn't be so strict about validation. If ever implemented, maybe it's a good idea to provide an option to turn validation off.

Okay, syntax check could come any time :)

**Comment by lezsakdomi** [comment]
Oh, and I was able to test my program using WinSCP - the only client implementing HOST I found so far
